### ISSUE TITLE
RouterExtensions memory footprint increases significantly compared to *ngIf, possible memory leak!

### ISSUE THREAD

**Issue by NorbertOcskai** [NativeScript/NativeScript#9972]
### Issue Description

Here is a demo for reproduction [https://github.com/palikollar/ns-routing-memory-leak](https://github.com/palikollar/ns-routing-memory-leak)

There are 4 routing types implemented there. Just comment out the type what you want to try out / reproduce.

1. "if routing"
Just two WebView component with *ngIf. The memory increased there ~20mb after 20 mins.
<img width="586" alt="if-routing" src="https://user-images.githubusercontent.com/102916788/180944047-ec26abc8-38c3-4f85-9763-629124c150b5.png">

2. back navigation
Navigates to nested routes and then back(). The memory increased ~50mb after 20 minutes.
<img width="584" alt="back-navigation" src="https://user-images.githubusercontent.com/102916788/180944078-97fb0365-ec75-45c7-ab5f-6fbd3df33853.png">

3. clear history navigation
Navigates to nested routes with clearHistory option set to true. The memory increased ~60mb after 20 minutes.
<img width="587" alt="clear-history" src="https://user-images.githubusercontent.com/102916788/180944098-b12f9494-430b-43bb-8d14-7cfe95eecb96.png">

4. Simple navigation
We can ignore this case because this is the incorrect way of using mobile stack navigation.
<img width="583" alt="simple-navigation" src="https://user-images.githubusercontent.com/102916788/180944193-98272ce9-efad-4c2e-8f28-8ae01ee48d26.png">

Why does the RouterExtensions navigate increased the memory footprint significantly compared to the simple *ngIf?

You can find the result screenshots here: [https://github.com/palikollar/ns-routing-memory-leak/tree/master/results](https://github.com/palikollar/ns-routing-memory-leak/tree/master/results)

### Reproduction

use the link demo

### Environment

OS: Windows 10 10.0.19044
CPU: (4) x64 Intel(R) Core(TM) i5-4670 CPU @ 3.40GHz
Shell: Not Found
node: 16.15.1
npm: 8.11.0
nativescript: Not Found

# android
java: 11.0.12
ndk: Not Found
apis: 28, 31, 32, 33
build_tools: 28.0.0, 28.0.3, 30.0.3, 33.0.0
system_images:
  - android-30 | Google APIs Intel x86 Atom
  - android-31 | Google APIs Intel x86 Atom_64

# ios
xcode: Not Found
cocoapods: Not Found
python: Not Found
python3: Not Found
ruby: Not Found
platforms: Not Found

### Dependencies

"dependencies": {
  "@angular/animations": "~14.1.0",
  "@angular/common": "~14.1.0",
  "@angular/compiler": "~14.1.0",
  "@angular/core": "~14.1.0",
  "@angular/forms": "~14.1.0",
  "@angular/platform-browser": "~14.1.0",
  "@angular/platform-browser-dynamic": "~14.1.0",
  "@angular/router": "~14.1.0",
  "@nativescript/angular": "^14.0.0",
  "@nativescript/core": "~8.2.0",
  "@nativescript/theme": "~3.0.2",
  "rxjs": "~7.5.0",
  "zone.js": "~0.11.5"
},
"devDependencies": {
  "@angular-devkit/build-angular": "~14.1.0",
  "@angular/compiler-cli": "~14.1.0",
  "@nativescript/android": "8.2.4",
  "@nativescript/types": "~8.2.0",
  "@nativescript/webpack": "~5.0.6",
  "@ngtools/webpack": "~14.1.0",
  "typescript": "~4.7.0"
}

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

**Comment by NathanWalker** [comment]
@NorbertOcskai thanks for the report, could you bump the following to 8.3.0:
```
"@nativescript/core": "~8.3.0",

"@nativescript/android": "~8.3.0",
"@nativescript/ios": "~8.3.0",
```
Then `ns clean` .... if you don't see immediate improvement it likely could be related to this which we are doing some testing with at moment: https://github.com/NativeScript/NativeScript/pull/9973

**Comment by NathanWalker** [comment]
This was also just merged into `next` and if you ensured you had these changes could also help your case:
https://github.com/NativeScript/NativeScript/pull/9977

**Comment by NathanWalker** [comment]
@NorbertOcskai when possible could you let us know if using `@nativescript/core: "next"` helped your case here?

**Comment by NathanWalker** [comment]
Confirmed fix in >=8.3.2
